packages/eslint-plugin/lib/config/rules/react.js

```diff
@@ -129,7 +133,15 @@ module.exports = {
   // Validate JSX indentation
   'react/jsx-indent': ['error', 2],
   // Validate JSX has key prop when in array or iterator
-  'react/jsx-key': 'error',
+  // checkFragmentShorthand - When true the rule will check if usage of the shorthand fragment syntax requires a key.
+  // checkKeyMustBeforeSpread - When true the rule will check if key prop after spread to avoid createElement fallback.
+  'react/jsx-key': [
+    'error',
+    {
+      checkFragmentShorthand: true,
+      checkKeyMustBeforeSpread: true,
```

Comment:
We want people to override keys?

Reply:
We want to prevent them from overriding a key. To prevent createElement fallback as seen here:
https://github.com/facebook/react/issues/20031#issuecomment-710346866